src/ripple/app/tx/impl/Payment.cpp

```diff
@@ -269,6 +269,14 @@ Payment::preclaim(PreclaimContext const& ctx)
 
         return tecDST_TAG_NEEDED;
     }
+    else if (sleDst->getFlags() & lsfAMM)
+    {
+        // Paying directly into the AMM pool is invalid.
```

Comment:
We need to disable payment channels and escrows as well, don't we? I understand why a regular account is used for AMMs, but this shows how adding a new feature causes complexity when interacting with old features. Any future transactions that can move funds needs to remember to disable it for AMM accounts too. This is error prone. What are the consequences if we mistakenly allow this?